docs/dev/software_development_process.md

```diff
@@ -0,0 +1,109 @@
+# USBNugget software development process
+
+## 1. Work planning and task creation
+
+For now, we will decide what work needs to be done during our weekly meeting.
+Work that needs doing is listed on [this project's Kanban
+board](https://github.com/orgs/HakCat-Tech/projects/4) under the "To do"
+column. Work that needs doing but is not fully defined can be added to the
+"Backlog" column.
+
+## 2. Communicate intended implementation
+
+Extremely trivial changes require no communication, other than updating the
+Kanban board to reflect what you'll be working on.
+
+When implementing non-trivial changes, share your ideas before implementing
+them.
+- For minor features, a short discussion during regular meeting or over chat is
+  probably sufficient
+- For major features, strongly consider creating a design document. This allows
+  other developers to comment on the design, suggest improvements or
+  alternatives, and think about edge cases. This prevents wasting time
+  constructing flawed implementations, and provides future developers with
+  artifacts to understand the code base and past decisions. Design documents
+  should be as short as possible while fully capturing your ideas. Use Google
+  docs or some other service that allows others to easily add comments in
+  specific sentences/paragraphs.
+
+## 3. Implementation
+
+### Style guide
+Overly strict adherence to a style guide is unproductive. If there is
+disagreement between two valid methods of implementation, a style guide can
+serve as an unopinionated third-party to resolve the conflict. Code should
+loosely follow the guide, and reviewers can consult the guide to suggest code
+improvements.
+
+For this project, we will be using the [Google C++ style
+guide](https://google.github.io/styleguide/cppguide.html). If we find that it
+doesn't meet our needs, we can copy and modify it or choose another entirely.
+
+### Use git commit best practices
+See [this
+gist](https://gist.github.com/luismts/495d982e8c5b1a0ced4a57cf3d93cf60) for
+some best practices. In general, things can be as messy as you want during
+development, but clean up before submitting a PR. Break large commits into
+smaller ones when it's sensible.
+
+### Documentation
```

Comment:
Separating user docs & dev docs is a good idea - we might consider making a simple static site doc generator w/ GitHub pages at some point to keep the readme less cluttered.